test/e2e/unmanaged_test.go

```diff
@@ -0,0 +1,47 @@
+package e2e_test
+
+import (
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	consoleapi "github.com/openshift/console-operator/pkg/api"
+	"github.com/openshift/console-operator/pkg/testframework"
+)
+
+// TestUnmanaged sets console-operator to Unmanaged state. After that "openshift-console
+// deployment is deleted after which the deploymnet is tested for unavailability, to
+// check that it wasn't recreated byt the console operator. Other resources from the
+// 'openshift-console' namespace (ConfigMap, Router, Service) are tested for availability
+// since they have not been deleted.
+func TestUnmanaged(t *testing.T) {
+	client := testframework.MustNewClientset(t, nil)
+
+	defer testframework.MustManageConsole(t, client)
+
+	testframework.MustUnmanageConsole(t, client)
+
+	t.Logf("deleting console deployment...")
+	if err := testframework.DeleteCompletely(
+		func() (metav1.Object, error) {
+			return client.Deployments(consoleapi.OpenShiftConsoleOperatorNamespace).Get(consoleapi.OpenShiftConsoleName, metav1.GetOptions{})
+		},
+		func(deleteOptions *metav1.DeleteOptions) error {
+			return client.Deployments(consoleapi.OpenShiftConsoleOperatorNamespace).Delete(consoleapi.OpenShiftConsoleName, deleteOptions)
+		},
+	); err != nil {
+		t.Fatalf("unable to delete console deployment: %s", err)
+	}
+
+	t.Logf("waiting to check if the operator has not recreate console deployment...")
+	errChan := make(chan error)
+	go testframework.IsResourceAvailable(errChan, client, "ConfigMap")
```

Comment:
Should we be asking for a named resources?  For example, we have 2 configmaps now.

Reply:
(other resources are singletons, though there is also more than 1 secret in the namespace)

Reply:
yeah, so here I wasn't sure if we want to nuke all the resources, or just some, or just one?
But since the operator should recreate any resource it should be enough to just test some resources. Thoughts ? :)

Reply:
Agree, though it might be good to know for certain it is recreating all of the expected resources.  If our [listOpts funcs](https://github.com/openshift/console-operator/blob/master/pkg/console/starter/starter.go#L72) are ever updated, it could be possible the operator would stop paying attention to a resource it needs to manage.  Currently the funcs are `noop` except for the `OAuth` client func.